### PR TITLE
improves testing instructions for dev preview

### PR DESCRIPTION
**Pull request by mhausenblas**
For the dev preview, make all steps explicit and ask for feedback

### CODE REVIEW

File: docs/contents/dev-docs/testing.md [hunk 1]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
-		"Effect": "Allow",
-		"Principal": {
-			"AWS": "arn:aws:iam::12345678970:user/kubernetes"
-		},
-		"Action": "sts:AssumeRole"
-	}
-}
-```
+From this above command's output, note the `Arn` field value, something along
+the lines of `arn:aws:iam::1234567890121:user/ausername` and store it in
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:136]: not sure if you meant the above to be "ausername" ?

File: docs/contents/dev-docs/testing.md [hunk 2]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+an environment variable called `ACK_PRINCIPAL_ARN`.
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:137]: How about `ACK_TEST_PRINCIPAL_ARN` to be consistent with the test role env var name?

File: docs/contents/dev-docs/testing.md [hunk 3]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+    If you're not that familiar with IAM access delegation, we recommend you
+    to peruse the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html)
 
-!!! note
-    We use [mockery](https://github.com/vektra/mockery) command for unit testing. You can install it by following [the guideline on the mockery's GitHub repository](https://github.com/vektra/mockery) or by just running our handy scirpt at `./scripts/install_mockery.sh` for general Linux environments.
+Next, in order for our test to generate [temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html)
+we need to tell it to use the IAM role we created in the previous step:
+
+```
+export AWS_ROLE_ARN=arn:aws:iam::$ACCOUNT_ID/$ACK_TEST_IAM_ROLE
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:182]: s/ACCOUNT_ID/AWS_ACCOUNT_ID/

File: docs/contents/dev-docs/testing.md [hunk 4]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+step.
 
-To clean up a `kind` Kubernetes cluster, which includes all the
-configuration files created by the script specifically for your test cluster,
-execute:
+!!! warning
+    If you try the following command and you see an error message containing
+    something along the line of `AWS_ROLE_ARN is not defined.` then you know
+    that somewhere in the IAM setup you either left out a step or one of the
+    commands failed.
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:204]: :+1:

File: docs/contents/dev-docs/testing.md [hunk 5]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+
+Now we're finally in the position to execute the end-to-end test:
+
+```
+make kind-cluster SERVICE=$SERVICE
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:209]: The above doesn't actually run tests. The `make kind-cluster-functional` target does, though, and you need to pass the ROLE_ARN, like so:

```
make kind-cluster-functional SERVICE=s3 ROLE_ARN=$AWS_ROLE_ARN
```

https://github.com/aws/aws-controllers-k8s/blob/main/Makefile#L39

File: docs/contents/dev-docs/testing.md [hunk 6]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+the respective ACK service controller is created, updated and deleted
+appropriately (still TODO).
+
+Finally, it will run tests that create resources for the respective service
+and verify if the resource has successfully created. In our example case it
+should create an S3 bucket and then destroy it again, yielding something like
+the following (edited down to the relevant parts):
+
+```
+```
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:229]: Think you're missing something here :)

File: docs/contents/dev-docs/testing.md [hunk 7]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+
+!!! tip "Keeping the test cluster around"
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:231]: By default, `make kind-cluster-functional` actually preserves the Kind cluster.

File: docs/contents/dev-docs/testing.md [hunk 8]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+    By default the script deletes the `kind` cluster. You can prevent this last
+    step from happening by usign `make kind-cluster-preserve SERVICE=$SERVICE`.
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:233]: I would just remove the above...

File: docs/contents/dev-docs/testing.md [hunk 9]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+Linux environments.
+
+
+ We track testing in the umbrella [issue 6](https://github.com/aws/aws-controllers-k8s/issues/6).
+ on GitHub. Use this issue as a starting point and if you create a new
+ testing-related issue, mention it from there.
+
+## Clean up
+
+If you used the `make kind-cluster-preserve` command, the `kind` cluster (Docker
```
Comment (jaypipes) [inline docs/contents/dev-docs/testing.md:249]: We should just remove the `kind-cluster-preserve` make target IMHO. The `kind-cluster-functional` target already preserves the test cluster.

File: docs/contents/dev-docs/testing.md [hunk 10]
```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
...
+In the following, we assume that you know your AWS account ID. Let's say it's 
+`1234567890121` (replace with your own in the following):
 
-Then, the above script runs a series of bash test scripts that call `kubectl`
-and the `aws` CLI tools to verify that custom resources of the type managed by
-the respective ACK service controller is created, updated and deleted
-appropriately (still TODO).
+```
+export ACCOUNT_ID=1234567890121
+```
```
[the remark this answers is not in the document]
Comment (mhausenblas) [reply docs/contents/dev-docs/testing.md:118]: I think' I'll simplify it even further using `aws sts get-caller-identity --query 'Account' --output text` directly.

File: docs/contents/dev-docs/testing.md (hunk 5 above)
Comment (mhausenblas) [reply docs/contents/dev-docs/testing.md:209]: not needed anymore

File: docs/contents/dev-docs/testing.md (hunk 9 above)
Comment (mhausenblas) [reply docs/contents/dev-docs/testing.md:249]: Once we've cleaned up the Makefile yes. Let's only document what currently exists/works, not aspirational stuff ;)

**Pull request merged**